app/workers/deletes.php

```diff
@@ -95,6 +120,70 @@ protected function deleteUser(Document $document, $projectId)
         ], $this->getProjectDB($projectId));
     }
 
+    protected function deleteExecutionLogs(Document $document) 
+    {
+        $projectIds = $document->getAttribute('projectIds', []);
+        foreach ($projectIds as $projectId) {
+            if (!($projectDB = $this->getProjectDB($projectId))) {
+                throw new Exception('Failed to get projectDB for project '.$projectId, 500);
+            }
+
+            // Delete Executions
+            $this->deleteByGroup([
+                '$collection='.$document->getCollection(),
+                '$projectId='.$projectId
+            ], $projectDB);
+        }
+    }
+
+    protected function deleteAbuseLogs($document) 
+    {
+        global $register;
+        $projectIds = $document->getAttribute('projectIds', []);
+        $timestamp = $document->getAttribute('timestamp', 0);
+
+        if($timestamp == 0) {
+            throw new Exception('Failed to delete abuse logs. No timestamp provided');
+        }
+
+        $timeLimit = new TimeLimit("", 0, 1, function () use ($register) {
+            return $register->get('db');
+        });
+        
+        foreach ($projectIds as $projectId) {
+            Console::success("Deleting abuse logs for Project: ", $projectId);
+            $timeLimit->setNamespace('app_'.$projectId);
+            $abuse = new Abuse($timeLimit); 
+
+            $status = $abuse->cleanup($timestamp);
+            if (!$status) {
+                throw new Exception('Failed to delete Abuse logs for project '.$projectId, 500);
```

Comment:
You don't need HTTP related error codes here, because this is not the HTTP API scope. You can just skip that param.